hydra-node/test/Hydra/Chain/Direct/Contract/Mutation.hs

```diff
@@ -591,6 +592,20 @@ changeMintedValueQuantityFrom tx exclude =
  where
   mintedValue = txMintValue $ txBodyContent $ txBody tx
 
+-- | A 'Mutation' that changes the forged quantity of tokens like this:
+-- - when no value is being forged -> add some value to be forged
+-- - when tx is forging values -> add more values on top of that
+changeMintedTokens :: Tx -> Value -> Gen Mutation
+changeMintedTokens tx mintValue =
+  ChangeMintedValue
+    <$> case mintedValue of
+      TxMintValueNone ->
+        pure mintValue
+      TxMintValue v _ ->
+        pure $ v <> mintValue
+ where
+  mintedValue = txMintValue $ txBodyContent $ txBody tx
```

Comment:
This is a very generic interface. Maybe `mintToken` or `burnToken` is what you would like to express more often in our tests? The arguments of such functions would be `PolicyId -> TokenName -> Natural` and it would be clear that we are burning 1 token when giving `1`, where in fact the quantity `-1` would be added to the `Value` in the field.

Reply:
I added a generator for the values that should produce either mined or burned one in the `genMintedOrBurnedValue`. If we still want to be more specific I could use `Quantity` as the argument here to be explicit.